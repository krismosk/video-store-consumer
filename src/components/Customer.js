import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './RentalCard.css';

const Customer = ({ id, name, address, city, state, postalCode, phone, findCustomer }) => {
  
  return (
  <div>
    <section>
      <p>{ id }</p>
      <p>{ name }</p>
      <p>{ address }</p>
      <p>{ city }</p>
      <p>{ state }</p>
    </section>
    <button onClick={() => {findCustomer(id)}} type="button">Select</button>
  </div>
  )
}

Customer.propTypes = {
  findCustomer: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.string,
  city: PropTypes.string,
  state: PropTypes.string,
  postalCode: PropTypes.string,
  phone: PropTypes.string,
}

export default Customer;