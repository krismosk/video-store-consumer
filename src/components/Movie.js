import React, { Component } from 'react';

const Movie = (props) => {
  
  return(
    <section>
      <img src={props.image} alt={`${props.title}`}/>
      <p> {props.title} </p>
      <p> {props.overview} </p>
      <p> {props.releaseDate} </p>
      <p> {props.inventory} </p>
      
      

      <button 
        type='button'
        onClick={() => {props.findMovie(props.id)}}
      >
      Select</button>
    </section>
  )
}

export default Movie;