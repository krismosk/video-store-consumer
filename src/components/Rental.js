import React, { Component } from 'react';
import axios from 'axios';

let date = new Date();
date.setDate(new Date().getDate() + 7);

class Rental extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      rental: '',
      dueDate: date,
    };
  }

  createRental = (movie, customer, dueDate) => {
    let params = {
      customer_id: customer.id,
      movie_id: movie.id,
      due_date: dueDate.toISOString(),
    }

    axios.post(`http://localhost:3000/rentals/${movie.title}/check-out`, params)
    .then((response) => {
      this.setState({rental: response.data})
    })
    .catch((error) => {
      this.setState({ error: error.message });
    });
  }

  showRental = () => {
    const showMovie = this.props.movie ? <p>{this.props.movie.title}</p> : "";
    const showCustomer = this.props.customer ? <p>{this.props.customer.name}</p> : "";

    return (
      <div>
        <p>{showMovie}</p>
        <p>{showCustomer}</p>
        <p>{`${this.state.rental}`}</p>
        <button onClick={() => {this.createRental(this.props.movie, this.props.customer, this.state.dueDate)}}>Checkout Rental</button>
      </div>
    )
  }

  render() {
    const rentalContent = this.props.movie || this.props.customer ? <p>{ this.showRental() }</p> : "";
    
    return (
    <div>
      <section>
        { rentalContent }
      </section>
    </div>
    )
  }
}

export default Rental;