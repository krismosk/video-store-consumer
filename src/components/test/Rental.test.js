import React from 'react';
import { render, cleanup, getByText } from '@testing-library/react';
import Rental from '../Rental';
import Movie from '../Movie'
import Customer from '../Customer'

const movie = <Movie 
  id={ 1337 }
  title={'test title'}
  image={'test image'}
  overview={'test overview'}
  releaseDate={'test date'}
  inventory={1}
  findMovie={() => { }}
/>;

const customer = <Customer 
  key={ 1 }
  id={ 1337 }
  name={ 'test customer' }
  address={ 'test address' }
  city={ 'test city' }
  state={ 'test state' }
  postalCode={ 'test postalcode' }
  phone={ '1111111111' }
  findCustomer={ () => { } }
/>;

const rental = <Rental
  movie={movie}
  customer={customer}
/>;

describe('Rental', () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render( rental );
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});