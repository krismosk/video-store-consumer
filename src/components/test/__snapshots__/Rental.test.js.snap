// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rental it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <section>
      <p />
      <p />
      <p />
      <button>
        Checkout Rental
      </button>
    </section>
  </div>
</DocumentFragment>
`;
